### PR TITLE
Changed max concurrent imports and use .limits.xxx instead

### LINKED ISSUE #5117: remove fixed number of max concurrent imports and use .limits.xxx instead
Now that we expose `user_data.limits.import_file_size` we can get rid of the frontend "magic number" of max concurrent imports that was previously hardcoded.

You can also check the [user_decorator](https://github.com/CartoDB/cartodb/blob/master/app/models/user/user_decorator.rb#L68) for all those names.

### PR DESCRIPTION
**Pull request by xavijam**
Fixes #5117.

Reviewer for frontend: @viddo

### CODE REVIEW

**Comment by xavijam** [comment]
Your turn @Kartones :)

**Comment by viddo** [comment]
:+1: although why prefix the names with get when the convention seems to be that we don't do that?

**Comment by Kartones** [comment]
@juanignaciosl CR of the BE code please

File: lib/assets/javascripts/cartodb/models/user.js [hunk 1]
```diff
@@ -103,6 +103,14 @@ cdb.admin.User = cdb.core.Model.extend({
     return true
   },
 
+  getMaxLayers: function() {
+    return ( this.get('limits') && this.get('limits').max_layers ) || 5;
+  },
+
+  getMaxConcurrentImports: function() {
+    return ( this.get('limits') && this.get('limits').concurrent_imports ) || 1;
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/models/user.js]: Why not provide a value object like `userLimits` that encapsulates business logic (e.g. fallbacks/defaults) related to the limits? What do you think about something like this?
```js
var userLimits = user.limits();
userLimits.maxLayers() // => 5
userLimits.maxConcurrentImports() // => 1
```

File: lib/assets/javascripts/cartodb/table/actions_menu.js [hunk 2]
```diff
@@ -40,6 +40,12 @@
       this.user = this.options.user;
       this.globalError = this.options.globalError;
 
+      // Set max layers if user has this parameter
+      var userMaxLayers = this.user.getMaxLayers();
+      if (!isNaN(userMaxLayers) && this._MAX_LAYERS !== userMaxLayers) {
+        this._MAX_LAYERS = userMaxLayers;
+      }
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: What's the purpose of duplicating user state onto this view? From what I can see this is only used in a test, if it's for assertion that should be solvable in the test itself. Also, how do these undefined/null checks make sense if the getters now return default values?

File: lib/assets/javascripts/cartodb/models/user.js (hunk 1 above)
Comment (Kartones) [inline lib/assets/javascripts/cartodb/models/user.js]: @xavijam is the one responsible for the FE code, I just exposed the stuff at user decorator/presenter.

PS: But I like that

File: lib/assets/javascripts/cartodb/table/actions_menu.js (hunk 2 above)
Comment (Kartones) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: I just merged master, so it's already there

**Comment by juanignaciosl** [comment]
LGTM :+1:

File: lib/assets/javascripts/cartodb/table/actions_menu.js (hunk 2 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: wait, there has been an error in the merge!

Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: Nobody added that, in fact I did a change yesterday where I removed that code :S

File: lib/assets/javascripts/cartodb/models/user.js (hunk 1 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/models/user.js]: I like your idea, but I feel we can't do it in this PR, what don't we create a new issue with that proposal?

File: lib/assets/javascripts/cartodb/table/actions_menu.js (hunk 2 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: Check master -> https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb/table/actions_menu.js#L43

Comment (Kartones) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: Then please take a look at the [merge commit](https://github.com/CartoDB/cartodb/commit/54890968038d9ece34d0f9b64e9aca1d741fafa6) for possible JS issues, I had a few conflicts and failing specs before the merge (due to a wrong default max layers value).
I did my best but...

File: lib/assets/javascripts/cartodb/models/user.js (hunk 1 above)
Comment (viddo) [inline lib/assets/javascripts/cartodb/models/user.js]: done, #5455

File: lib/assets/javascripts/cartodb/table/actions_menu.js (hunk 2 above)
Comment (Kartones) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: fixed that issue

**Comment by xavijam** [comment]
Much appreciated

File: lib/assets/javascripts/cartodb/table/actions_menu.js (hunk 2 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: Thanks

**Comment by Kartones** [comment]
@xavijam Please take a look at the mentined diff because removal of those lines weren't supposed to be there has broken the frontend tests :/
http://ci.cartodb.com/job/CartoDB-PR-testing/756/consoleText

**Pull request merged**